src/tests/MetaDataQueryTests.cpp

```diff
@@ -0,0 +1,207 @@
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This code has been produced by the European Spallation Source
+// and its partner institutes under the BSD 2 Clause License.
+//
+// See LICENSE.md at the top level for license information.
+//
+// Screaming Udder!                              https://esss.se
+
+#include "KafkaW/MetaDataQueryImpl.h"
+#include "helpers/KafkaWMocks.h"
+#include <gtest/gtest.h>
+
+class CreateKafkaHandleTest : public ::testing::Test {};
+
+TEST_F(CreateKafkaHandleTest, Success) {
+  auto Result =
+      KafkaW::getKafkaHandle<RdKafka::Consumer, RdKafka::Conf>("some_broker");
+  EXPECT_FALSE(Result == nullptr);
+}
+
+class UsedProducerMock {
+public:
+  static int CallsToCreate;
+  static RdKafka::Producer *create(RdKafka::Conf *, std::string &) {
+    UsedProducerMock::CallsToCreate++;
+    return nullptr;
+  }
+};
+int UsedProducerMock::CallsToCreate = 0;
+
+TEST_F(CreateKafkaHandleTest, FailedToCreateHandle) {
+  UsedProducerMock::CallsToCreate = 0;
+  auto testFunc = [](auto Adr) { // Work around for GTest limitation
+    return KafkaW::getKafkaHandle<UsedProducerMock, RdKafka::Conf>(Adr);
+  };
+  EXPECT_THROW(testFunc("some_broker"), MetadataException);
+  EXPECT_EQ(UsedProducerMock::CallsToCreate, 1);
```

Comment:
Not sure I see the point of this.

Reply:
That is fine. You do not have to. 😃 It is simply the result of limitations in mocking in C++ when using free functions.